romancal/resample/resample.py

```diff
@@ -732,30 +735,69 @@ def gwcs_into_l3(model, wcsinfo):
     by indexing. This is fragile and will be a source of issues.
     """
     l3_wcsinfo = model.meta.wcsinfo
-    transform = wcsinfo.forward_transform
+    transform = wcs.forward_transform
 
-    # Basic WCS info
     l3_wcsinfo.projection = "TAN"
-    l3_wcsinfo.rotation_matrix = transform["pc_rotation_matrix"].matrix.value.tolist()
-    l3_wcsinfo.dec_ref = transform.lat_6.value
-    l3_wcsinfo.ra_ref = transform.lon_6.value
-    # l3_wcsinfo.x_ref = center of mosaic?
-    # l3_wcsinfo.y_ref = center of mosaic?
-    # l3_wcsinfo.s_region = from gwcs.bounding_box
-
-    # Mosaic info
-    # l3_wcsinfo.dec_center = from center of mosaic
-    # l3_wcsinfo.pixel_scale = ???
-    # l3_wcsinfo.pixel_scale_local = ???
-    # l3_wcsinfo.pixel_shape = image shape
-    # l3_wcsinfo.ra_center = from center of mosaic
-    # l3_wcsinfo.ra_corn1 = from bounding box
-    # l3_wcsinfo.ra_corn2 = from bounding box
-    # l3_wcsinfo.ra_corn3 = from bounding box
-    # l3_wcsinfo.ra_corn4 = from bounding box
-    # l3_wcsinfo.dec_corn1 = from bounding box
-    # l3_wcsinfo.dec_corn2 = from bounding box
-    # l3_wcsinfo.dec_corn3 = from bounding box
-    # l3_wcsinfo.dec_corn4 = from bounding box
-    # l3_wcsinfo.orientat = ???
-    # l3_wcsinfo.orientat_local = ???
+    l3_wcsinfo.pixel_shape = model.shape
+
+    world_center = wcs(*[(v - 1) / 2.0 for v in model.shape[::-1]])
+    l3_wcsinfo.ra_center = world_center[0]
+    l3_wcsinfo.dec_center = world_center[1]
+    l3_wcsinfo.pixel_scale_local = compute_scale(wcs, world_center)
+    l3_wcsinfo.orientat_local = calc_pa(wcs, *world_center)
+
+    try:
+        footprint = utils.create_footprint(wcs, model.shape)
+    except Exception as excp:
+        log.warning("Could not determine footprint due to %s", excp)
+    else:
+        l3_wcsinfo.ra_corn1 = footprint[0][0]
+        l3_wcsinfo.ra_corn2 = footprint[1][0]
+        l3_wcsinfo.ra_corn3 = footprint[2][0]
+        l3_wcsinfo.ra_corn4 = footprint[3][0]
+        l3_wcsinfo.dec_corn1 = footprint[0][1]
+        l3_wcsinfo.dec_corn2 = footprint[1][1]
+        l3_wcsinfo.dec_corn3 = footprint[2][1]
+        l3_wcsinfo.dec_corn4 = footprint[3][1]
+        l3_wcsinfo.s_region = utils.create_s_region(footprint)
+
+    try:
+        l3_wcsinfo.rotation_matrix = transform[
+            "pc_rotation_matrix"
+        ].matrix.value.tolist()
+        l3_wcsinfo.dec_ref = transform.lat_6.value
+        l3_wcsinfo.ra_ref = transform.lon_6.value
+        l3_wcsinfo.x_ref = -transform["crpix1"].offset.value
+        l3_wcsinfo.y_ref = -transform["crpix2"].offset.value
```

Comment:
Along the same lines as Mairan, is it the case that we only need to depend on one of either
crpix1, crpix2, or
lat_6, lon_6
existing?  I guess I'd prefer the former and get the latter two via wcs.pixel_to_world(x_ref, y_ref) if that doesn't throw an out-of-bounds error?

Reply:
No, it's not guaranteed to always have lat_6 or lon_6. The number attached to the end comes from the order of the models. It's better to index the compound transform with the model name to retrieve the parameters.
 I commented above on the correct way to get to the values but those comments were resolved without addressing them.

Reply:
Yes, needed to ask about that. That transform does not have a name and will need to be added. Any suggestions on a name?

FYI: the model currently appears as follows:
```
Components:
    [0]: <Shift(offset=-118.80708394, name='crpix1')>

    [1]: <Shift(offset=-90.02345166, name='crpix2')>

    [2]: <AffineTransformation2D(matrix=[[-0.93358043,  0.35836795], [ 0.35836795,  0.93358043]], translation=[0., 0.], name='pc_rotation_matrix')>

    [3]: <Scale(factor=0.00002778, name='cdelt1')>

    [4]: <Scale(factor=0.00002778, name='cdelt2')>

    [5]: <Pix2Sky_Gnomonic()>

    [6]: <RotateNative2Celestial(lon=10.0029245, lat=0.0015345, lon_pole=180.)>
```
and
```
>>> print(wcs.forward_transform.submodel_names)
('crpix1', 'crpix2', 'pc_rotation_matrix', 'cdelt1', 'cdelt2', 'None_0', 'None_1')
```

Reply:
I think if crpix1 & crpix2 work, we can get lat_6 and lon_6 by wcs.pixel_to_world(x_ref, y_ref) in a generic way, so that solves the issue?